src/services/rickmorty.service.mjs

```diff
@@ -0,0 +1,76 @@
+import axios from "axios";
+
+const api = axios.create({
+  baseURL: "https://rickandmortyapi.com/graphql",
+});
+
+const MAX_PAGE = Math.ceil(671 / 20);
+const MAX_ITEMS_PER_PAGE = 20;
+
+function randomPage(maxPage) {
+  return Math.floor(Math.random() * maxPage + 1);
+}
+
+function buildQuery() {
+  const query = `
+  query {
+    characters(page: ${randomPage(MAX_PAGE)}) {
+      info {
+        count
+      }
+      results {
+        name,
+        image
+      }
+    }
+  }`;
+  return query;
+}
+
+function createCardObject(character, index) {
+  if (!character) throw new Error("Character is undefined");
+  return {
+    id: index + 1,
+    name: character.name,
+    image: character.image,
+  };
+}
+
+async function getCards(amount) {
+  try {
+    if (!amount || amount <= MAX_ITEMS_PER_PAGE) {
+      const query = buildQuery();
+      const res = await api.post("/", { query });
+      return res.data.data.characters.results
+        .map((char, index) => createCardObject(char, index))
+        .slice(0, amount);
+    } else {
+      const pageQueries = [];
+
+      // call multiple pages if the card amount is bigger than MAX_ITEMS_PER_PAGE
+      for (let i = 0; i < Math.ceil(amount / MAX_ITEMS_PER_PAGE); i++) {
+        const query = buildQuery();
```

Comment:
Du musst ausschließen, dass du die gleiche query mehrmals buildest.

Reply:
Wieso? Ich mach den query ja random immer bei jedem game init. Die Wahrscheinlichkeit das zweimal hintereinander die selbe query/page kommt ist denke ich sehr gering, bzw vernachlässigbar.

Reply:
Die Wahrscheinlichkeit hängt von der maximalen Anzahl der Pages und der angefragten Anzahl der Pages ab. Die Kollisionswahrscheinlichkeit beträgt bei ner API mit 73 Pages schon 5% wenn ich nur 2 Pages anfordere. Das heißt schon nach 13 Spielen ist es wahrscheinlicher, dass ich den Bug hatte, als dass ich ihn nicht hatte.
Wichtiger ist allerdings die Auswirkung des Bugs. Der wäre gamebreaking, denn ich hätte plötzlich Karten auf dem Gameboard, die optisch identisch sind, aber nicht als Pärchen gelten.

Reply:
added as todo in trelleo